Libraries/Opc.Ua.Server/Diagnostics/CustomNodeManager.cs

```diff
@@ -945,20 +930,17 @@ protected virtual NodeHandle GetManagerHandle(ServerSystemContext context, NodeI
                 return null;
             }
 
-            if (m_predefinedNodes != null)
-            {
-                NodeState node = null;
+            NodeState node = null;
 
-                if (m_predefinedNodes.TryGetValue(nodeId, out node))
-                {
-                    NodeHandle handle = new NodeHandle();
+            if (m_predefinedNodes?.TryGetValue(nodeId, out node) == true)
```

Comment:
Is this thread Safe w.o. lock and standard dictionary?

Reply:
We are only reading here, so this should apply:
> A [Dictionary<TKey,TValue>](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2?view=net-8.0) can support multiple readers concurrently, as long as the collection is not modified.

[Reference](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2?view=net-8.0&redirectedfrom=MSDN#thread-safety)

Reply:
This lock may not be safe to remove, unless we can be assured that the dictionary is not modified, I will check this again. The function causes a lot of contention in a loaded server.

Reply:
added the lock for the external facing GetManagerHandle back in, in a subsequent PR NodeIdDictionary could be transitioned to an underlying ConcurrentDictionary...